Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -522,6 +523,17 @@ protected override void PrepareContainerForItemOverride(DependencyObject element
             {
                 carouselItem.IsSelected = true;
             }
+
+            carouselItem.ParentCarousel = this;
```

Comment:
It looks like in the WinUI TabView they do a similar pattern but they use `weak_ref` in C++ to store the parent, should we do the same in C# here? @ranjeshj any guidance here?

Reply:
We can end up creating cycles in C++ when pointing to a parent from the child element, so we use weak refs for this case.

Reply:
Changed this now to be a weak reference 😄